packages/runtime/src/modules/DeciderModule.ts

```diff
@@ -22,18 +76,269 @@ export class DeciderModule extends RuntimeModule {
+        return requestConfigDates.every((requestConfigDate) => this.checkDateCompatibility(requestConfigDate, requestDate));
+    }
+
+    private checkDateCompatibility(requestConfigDate: string, requestDate: string): boolean {
+        if (requestConfigDate.startsWith(">")) return CoreDate.from(requestDate).isAfter(CoreDate.from(requestConfigDate.substring(1)));
+        if (requestConfigDate.startsWith("<")) return CoreDate.from(requestDate).isBefore(CoreDate.from(requestConfigDate.substring(1)));
+        return CoreDate.from(requestDate).equals(CoreDate.from(requestConfigDate));
+    }
+
+    private checkRequestItemCompatibilityAndApplyResponseConfig(
+        itemsOfRequest: (RequestItemJSONDerivations | RequestItemGroupJSON)[],
+        parametersToDecideRequest: any,
+        requestConfigElement: RequestItemDerivationConfig,
+        responseConfigElement: ResponseConfig
+    ): { items: any[] } {
+        for (let i = 0; i < itemsOfRequest.length; i++) {
+            const item = itemsOfRequest[i];
+            if (item["@type"] === "RequestItemGroup") {
+                this.checkRequestItemCompatibilityAndApplyResponseConfig(
+                    (item as RequestItemGroupJSON).items,
+                    parametersToDecideRequest.items[i],
+                    requestConfigElement,
+                    responseConfigElement
+                );
+            } else {
+                const alreadyDecidedByOtherConfig = !!parametersToDecideRequest.items[i];
+                if (alreadyDecidedByOtherConfig) continue;
+
+                if (isRequestItemDerivationConfig(requestConfigElement)) {
+                    const requestItemIsCompatible = this.checkRequestItemCompatibility(requestConfigElement, item as RequestItemJSONDerivations);
+                    if (!requestItemIsCompatible) continue;
+                }
+
+                if (isGeneralRequestConfig(requestConfigElement) && responseConfigElement.accept) {
+                    const requestItemsWithSimpleAccept = [
+                        "AuthenticationRequestItem",
+                        "ConsentRequestItem",
+                        "CreateAttributeRequestItem",
+                        "RegisterAttributeListenerRequestItem",
+                        "ShareAttributeRequestItem"
+                    ];
+                    if (!requestItemsWithSimpleAccept.includes(item["@type"])) continue;
+                }
+
+                parametersToDecideRequest.items[i] = responseConfigElement;
+            }
+        }
+        return parametersToDecideRequest;
+    }
+
+    public checkRequestItemCompatibility(requestConfigElement: RequestItemDerivationConfig, requestItem: RequestItemJSONDerivations): boolean {
+        const requestItemPartOfConfigElement = this.filterConfigElementByPrefix(requestConfigElement, true);
+        return this.checkCompatibility(requestItemPartOfConfigElement, requestItem);
+    }
```

Comment:
It feels like this code should not be inside the module itself - it's really blown up now.

Reply:
@jkoenig134 I see what you mean, however I'm not sure how to properly separate what should be part of the module and what should be done outside. E.g. why didn't you mark the function `createEmptyDecideRequestItemParameters`? Furthermore, would you put the part that should be moved outside of the module just below it in the same file?

Reply:
because the overview is a bit strange, I've should have marked that :)